test/unit_tests/service/grid/grid_service_test.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:f1_application/lib/model/driver.dart';
+import 'package:f1_application/lib/service/grid/grid_service.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:collection/collection.dart';
+
+void main() {
+  final GridService service = GridService();
+
+  final List<Driver> drivers = [
+    Driver(id: "1", wikiURL: "https://wikipedia.com", firstName: "One", lastName: "Driver", dateOfBirth: "2001-01-01", nationality: "Italian", code: "DON", permanentNumber: "01"),
+    Driver(id: "2", wikiURL: "https://wikipedia.com", firstName: "Two", lastName: "Driver", dateOfBirth: "2002-01-01", nationality: "Brazilian", code: "DTW", permanentNumber: "02"),
+    Driver(id: "3", wikiURL: "https://wikipedia.com", firstName: "Three", lastName: "Driver", dateOfBirth: "2003-01-01", nationality: "Columbian", code: "DTH", permanentNumber: "03"),
+    Driver(id: "4", wikiURL: "https://wikipedia.com", firstName: "Four", lastName: "Driver", dateOfBirth: "2004-01-01", nationality: "Hungarian", code: "DFO", permanentNumber: "04"),
+    Driver(id: "5", wikiURL: "https://wikipedia.com", firstName: "Five", lastName: "Driver", dateOfBirth: "2005-01-01", nationality: "Brazilian", code: "DFI", permanentNumber: "05"),
+  ];
+
+  group("Grid Service test(s)", () {
+    group("relevantDriversByExpression function test(s)", () {
+      group("With drivers List", () {
+        test("Get all driver with no expression ''", () {
+          final result = service.relevantDriversByExpression(drivers, "");
+          expect(const ListEquality().equals(result, drivers), true);

```

Comment:
Itt nem biztos, hogy az egész listára vizsgálnék, hanem csak a hosszukra. Gondolj bele abba, ha pl. rendezés is közrejátszana, esetleg bővülne a függvény paraméterlistája és funkcionalitása, akkor nem ugyanaz a lista fog már feltétlen visszajönni mint amit a bemeneten definiálsz. 

Nem nagy hiba és lehet hogy tökmindegy végső soron, de tapasztalatból itt ennyi elég lenne :)